lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -557,6 +560,25 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def confirm_ejson_keys_not_prunable
+      secret, err, st = ejson_provisioner.ejson_keys_secret
+      unless st.success? || err.include?(KubernetesDeploy::Kubectl::NOT_FOUND_ERROR_TEXT)
+        raise FatalDeploymentError,
+          "Error running validation for Secret/#{EjsonSecretProvisioner::EJSON_KEYS_SECRET}: #{err}"
+      end
+
+      if st.success?
```

Comment:
Is this check needed?

Reply:
Yes, since we might have a `NOT_FOUND` error on our hands instead of actual success.